logging/source/RialtoLogging.cpp

```diff
@@ -42,108 +42,154 @@ static const firebolt::rialto::logging::EnvVariableParser g_envVariableParser;
-    iov[0].iov_len = std::min(iov[0].iov_len, sizeof(tbuf));
+    switch (component)
+    {
+    case RIALTO_COMPONENT_DEFAULT:
+        return "DEF";
+        break;
+    case RIALTO_COMPONENT_CLIENT:
+        return "CLI";
+        break;
+    case RIALTO_COMPONENT_SERVER:
+        return "SRV";
+        break;
+    case RIALTO_COMPONENT_IPC:
+        return "IPC";
+        break;
+    case RIALTO_COMPONENT_SERVER_MANAGER:
+        return "SMG";
+        break;
+    case RIALTO_COMPONENT_COMMON:
+        return "COM";
+        break;
+    case RIALTO_COMPONENT_EXTERNAL:
+        return "EXT";
+        break;
+    default:
+        return "UNK";
+        break;
+    }
+}
 
+static std::string levelToString(RIALTO_DEBUG_LEVEL level)
+{
     switch (level)
     {
     case RIALTO_DEBUG_LEVEL_FATAL:
-        iov[1].iov_base = const_cast<void *>(reinterpret_cast<const void *>("FTL: "));
-        iov[1].iov_len = 5;
+        return "FTL";
         break;
     case RIALTO_DEBUG_LEVEL_ERROR:
-        iov[1].iov_base = const_cast<void *>(reinterpret_cast<const void *>("ERR: "));
-        iov[1].iov_len = 5;
+        return "ERR";
         break;
     case RIALTO_DEBUG_LEVEL_WARNING:
-        iov[1].iov_base = const_cast<void *>(reinterpret_cast<const void *>("WRN: "));
-        iov[1].iov_len = 5;
+        return "WRN";
         break;
     case RIALTO_DEBUG_LEVEL_MILESTONE:
-        iov[1].iov_base = const_cast<void *>(reinterpret_cast<const void *>("MIL: "));
-        iov[1].iov_len = 5;
+        return "MIL";
         break;
     case RIALTO_DEBUG_LEVEL_INFO:
-        iov[1].iov_base = const_cast<void *>(reinterpret_cast<const void *>("NFO: "));
-        iov[1].iov_len = 5;
+        return "NFO";
         break;
     case RIALTO_DEBUG_LEVEL_DEBUG:
-        iov[1].iov_base = const_cast<void *>(reinterpret_cast<const void *>("DBG: "));
-        iov[1].iov_len = 5;
+        return "DBG";
         break;
     case RIALTO_DEBUG_LEVEL_EXTERNAL:
+        return "EXT";
+        break;
     default:
-        iov[1].iov_base = const_cast<void *>(reinterpret_cast<const void *>(": "));
-        iov[1].iov_len = 2;
+        return ":";
         break;
     }
+}
+
+/**
+ * Console logging function for the library.
+ */
+static void consoleLogHandler(RIALTO_COMPONENT component, RIALTO_DEBUG_LEVEL level, const char *file, int line,
+                              const char *function, const char *message, size_t messageLen)
+{
+    timespec ts = {0, 0};
+    clock_gettime(CLOCK_MONOTONIC, &ts);
+    struct iovec iov[6];
+    char tbuf[32];
+
+    iov[0].iov_base = tbuf;
+    iov[0].iov_len = snprintf(tbuf, sizeof(tbuf), "%.010lu.%.06lu ", ts.tv_sec, ts.tv_nsec / 1000);
+    iov[0].iov_len = std::min(iov[0].iov_len, sizeof(tbuf));
+
+    char lbuf[8];
+    iov[1].iov_base = reinterpret_cast<void *>(lbuf);
+    iov[1].iov_len =
+        std::min(static_cast<size_t>(snprintf(lbuf, sizeof(lbuf), "%s: ", componentToString(component).c_str())),
```

Comment:
Should this be using levelToString() rather than componentToString()? It looks like a duplicate of what's sent in iov[2] at the moment.